src/frontend/Typechecker.ml

```diff
@@ -976,10 +976,10 @@ let verify_assignment_global loc cf block is_global id =
 (* Until function types are added to the user language, we
    disallow assignments to function values
 *)
-let verify_assignment_non_function loc ut id =
+let verify_assignment_non_function loc ut =
   match ut with
   | UnsizedType.UFun _ | UMathLibraryFunction ->
-      Semantic_error.cannot_assign_function loc ut id.name |> error
+      Semantic_error.cannot_assign_function loc ut |> error
   | _ -> ()
```

Comment:
Not good enough.
```stan
functions {
  int foo() {
    int x;
    (foo, x) = (foo, 1);
    return x;
  }
}
```
We might want to make function names always readonly and omit this type-based check.

Reply:
Making them readonly would work but doesn't produce as nice an error message. The fix here is easy enough that I think we can leave it as the type-based check